### PR TITLE
TCP: Allow connecting by hostname

### PR DESCRIPTION
**Pull request by ThomasDebrunner**
This allows TCP connections to be made by hostname, not just IP - e.g. `tcp://localhost:1234`, or `tcp://my-device.local:1234`

Tested on linux and macos

### CODE REVIEW

**Review by JonasVautherin — APPROVED** [review]
Nice! I fixed the style, and the other CI issues seem unrelated to this PR.

File: src/mavsdk/core/tcp_connection.cpp [hunk 1]
```diff
@@ -80,7 +81,16 @@ ConnectionResult TcpConnection::setup_port()
     struct sockaddr_in remote_addr {};
     remote_addr.sin_family = AF_INET;
     remote_addr.sin_port = htons(_remote_port_number);
-    remote_addr.sin_addr.s_addr = inet_addr(_remote_ip.c_str());
+
+    struct hostent* hp;
+    hp = gethostbyname(_remote_ip.c_str());
```
Comment (julianoes) [inline src/mavsdk/core/tcp_connection.cpp:86]:
```suggestion
    struct hostent* hp; = gethostbyname(_remote_ip.c_str());
```

As I read it, this doesn't need cleaning up, right?

**Review by julianoes — APPROVED** [review]
Any chance you can add the same for Windows while you're at it?

**Comment by julianoes** [comment]
CI likely doesn't test it. I'll try to test on Windows some time but you might beat me to it.

**Comment by julianoes** [comment]
@ThomasDebrunner sorry for the delay. Trying to build and run this on Windows caused some work :smile:, see #2069. So, anyway, let me just merge it.

**Pull request merged**